sdb/commands/exit.py

```diff
@@ -0,0 +1,32 @@
+#
+# Copyright 2019 Chuck Tuffli
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# pylint: disable=missing-docstring
+
+import sys
+from typing import Iterable
+
+import drgn
+import sdb
+
+
+class Echo(sdb.Command):
+    # pylint: disable=too-few-public-methods
+
+    names = ["exit", "quit"]
+
+    def call(self, objs: Iterable[drgn.Object]) -> Iterable[drgn.Object]:
+        sys.exit()
```

Comment:
Should we use 'raise SystemExit` instead? I'm not fluent in python enough to explain way the exception may be better, but I've seen the exception recommended in other code bases, so I figured I'd at least ask.

Reply:
Doing some superficial research it seems like `sys.exit()` and `raise SystemExit` do the same thing.

In my opinion though I think it would be preferable to `raise SystemExit` and then catch it in `sdb/internal/repl.py` here:
```
def start_session(self) -> None:
        """
        Starts a REPL session.
        """
        while True:
            try:
                line = input(self.prompt).strip()
            except (EOFError, KeyboardInterrupt, SystemExit): # <----- Added here
                print(self.closing)
                break

            if not line:
                continue
            _ = self.eval_cmd(line)
```

My reasoning for this has two points:
1] All the exit code paths (currently Ctrl+D and the `exit` command) go through the same closing code of the session of our REPL. This is helpful because, if we were to do add more cleanup functionality, you'd only have to care about this code path, not multiple codepaths (e.g. one for Ctrl+D and one for `exit`).
2] In the wild scenario when we have another consumer outside of the REPL (e.g. an SDB server that we contact over the network or whatever) we decouple that functionality by raising an Exception to exit and each consumer can handle exit however they want. Doing `sys.exit` and instantly halt the program may not be what each consumer as context may vary.

That said, I do find it annoying that we'd have to use an exception for something that is not very *exceptional* such as exiting the session. That said, I believe that the above benefits outweigh the drawbacks.

Reply:
I think the ability to catch the exit exception, if the consumer wishes to, is the main benefit of using it. Those two points specific to SDB make sense to me.

Reply:
I'm not a Python expert but checked with someone here, and they agreed that there would be a benefit to using `raise SystemExit` for the reason you both mentioned. I'll make that change and update a bit later.